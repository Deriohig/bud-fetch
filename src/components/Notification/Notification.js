import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const NotificationWrapper = styled.div`
    position: fixed;
    bottom: 16px;
    left: 16px;
    right: 16px;
    display: flex;
    background-color: #fff;
    height: 40px;
    border-left: 8px solid;
    border-top: 1px solid;
    border-bottom: 1px solid;
    border-right: 1px solid;
    border-color: ${(props) => (props.kind === 'error' ? '#da1e28' : 'white')};
    background-color: ${(props) => (props.kind === 'error' ? '#FEF0F0' : 'white')};
    cursor:pointer;
    line-height: 40px;
`;

const Title = styled.h5`
    font-weight: 600;
    display: inline-flex;
    font-size: 16px;
    margin-left: 8px;
`;

const Message = styled.p`
    font-size: 14px;
    display: inline-flex;
    font-weight: 500;
    margin-left: 8px;
`;

const Notification = ({ title, kind, message, handleClick }) => {
    return (
        <NotificationWrapper data-testid='Notification' onClick={() => handleClick()} kind={kind}>
            <Title>{title}</Title>
            <Message>{message}</Message>
        </NotificationWrapper>
    );
};

Notification.propTypes = {
    title: PropTypes.string.isRequired,
    kind: PropTypes.string.isRequired,
    message: PropTypes.string.isRequired,
    handleClick : PropTypes.func
}

export default Notification;
