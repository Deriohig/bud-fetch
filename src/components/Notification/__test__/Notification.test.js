import React from 'react';
import ReactDOM from 'react-dom';
import Notification from '../index';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer'

const props = {
    title: 'Title',
    kind: 'error',
    message: 'message',
};

afterEach(cleanup);

it("renders without crashing", ()=> {
    const div = document.createElement('div');
    ReactDOM.render(<Notification {...props} />, div)
})

it("renders with a title", ()=> {
    const { getByTestId } = render(<Notification {...props} />)
    expect(getByTestId('Notification')).toHaveTextContent('Title');
})

it("renders with a message", ()=> {
    const { getByTestId } = render(<Notification {...props} />)
    expect(getByTestId('Notification')).toHaveTextContent('message');
})

it("renders with an error style", ()=> {
    const { getByTestId } = render(<Notification {...props} />)
    const style = window.getComputedStyle(getByTestId('Notification'));
    expect(style.borderColor).toBe('#da1e28');
    expect(style.backgroundColor).toBe('rgb(254, 240, 240)');
})

it("matches snapshot", ()=> {
    const snapshot = renderer.create(<Notification {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
})
