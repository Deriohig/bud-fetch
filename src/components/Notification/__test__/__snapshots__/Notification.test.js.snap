// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bdfBwQ ffpIeZ"
  data-testid="Notification"
  kind="error"
  onClick={[Function]}
>
  <h5
    className="sc-gsTCUz hzcwwD"
  >
    Title
  </h5>
  <p
    className="sc-dlfnbm cXSvgV"
  >
    message
  </p>
</div>
`;
